frontend/src/component/releases/ReleasePlanTemplate/MilestoneCard.tsx

```diff
@@ -0,0 +1,109 @@
+import { StyledMilestoneCard, StyledMilestoneCardBody } from './TemplateForm';
+import Input from 'component/common/Input/Input';
+import { Button, Grid, styled } from '@mui/material';
+import Edit from '@mui/icons-material/Edit';
+import type { IReleasePlanMilestonePayload } from 'interfaces/releasePlans';
+import { useEffect, useState } from 'react';
+import { useTheme } from '@mui/material';
+
+const StyledEditIcon = styled(Edit)(({ theme }) => ({
+    cursor: 'pointer',
+    marginTop: theme.spacing(-0.25),
+    marginLeft: theme.spacing(0.5),
+    height: theme.spacing(2.5),
+    width: theme.spacing(2.5),
+    color: theme.palette.text.secondary,
+}));
+
+const StyledGridItem = styled(Grid)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'center',
+}));
+
+const StyledInput = styled(Input)(({ theme }) => ({
+    width: '100%',
+}));
+
+const StyledMilestoneCardTitle = styled('span')(({ theme }) => ({
+    fontWeight: theme.fontWeight.bold,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+interface IMilestoneCard {
+    index: number;
+    milestone: IReleasePlanMilestonePayload;
+    showAddStrategyDialog: (index: number) => void;
+    errors: { [key: string]: string };
+    clearErrors: () => void;
+}
+
+export const MilestoneCard = ({
+    index,
+    milestone,
+    showAddStrategyDialog,
+    errors,
+    clearErrors,
+}: IMilestoneCard) => {
+    const [milestoneName, setMilestoneName] = useState<string>(milestone.name);
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const theme = useTheme();
+
+    useEffect(() => {
+        milestone.name = milestoneName;
+    }, [milestoneName]);
```

Comment:
I believe this will cause issues. You should set the state using `setMilestones` instead. Ideally this component would have a `onNameChange` or similar method in their props and you would handle this in the MilestoneList. MilestoneList can accept the original setMilestones as a prop and use it inside the new `onNameChange` method to update the correct milestone in the list.

Reply:
I struggled finding a good abstraction for this as the list has a dynamic amount of elements and it all has to be saved as one big chunk. I'd love to get your thoughts/insights on that